src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs

```diff
@@ -69,7 +72,27 @@ public static TracerProviderBuilder SetResourceBuilder(this TracerProviderBuilde
         {
             if (tracerProviderBuilder is TracerProviderBuilderBase tracerProviderBuilderBase)
             {
-                tracerProviderBuilderBase.SetResourceBuilder(resourceBuilder);
+                tracerProviderBuilderBase.ResourceBuilder = resourceBuilder;
+            }
+
+            return tracerProviderBuilder;
+        }
+
+        /// <summary>
+        /// Modify the <see cref="ResourceBuilder"/> from which the Resource associated with
+        /// this provider is built from in-place.
+        /// </summary>
+        /// <param name="tracerProviderBuilder">TracerProviderBuilder instance.</param>
+        /// <param name="configure">An action which modifies the provided <see cref="ResourceBuilder"/> in-place.</param>
+        /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public static TracerProviderBuilder ConfigureResource(this TracerProviderBuilder tracerProviderBuilder, Action<ResourceBuilder> configure)
```

Comment:
Thoughts on the name `ConfigureResource` vs `ConfigureResourceBuilder`? @alanwest @Oberon00 ?